### DISCUSSION TITLE
smoothing kernels in specfem

### DISCUSSION THREAD

**Discussion by carltape** [SPECFEM/specfem3d#1555] [Ongoing Research]
Smoothing volumetric kernels can be computationally expensive. Many groups—salvus, U. Toronto, Princeton—are using PDE-based approaches (like diffusion equation) to smooth, rather than using convolution. UT and Princeton, are these tools in a state where they can be added into both packages, or are they still under development? It looks like the currently available approach for smoothing (and the one employed within seisflows/pyatoa) is convolution, as shown in [smooth_sem.F90](https://github.com/SPECFEM/specfem3d/blob/master/src/tomography/postprocess_sensitivity_kernels/smooth_sem.F90). Please chime in here if you are using these more efficient smoothing operations with specfem volumetric files. Thanks.

**Comment by rdno** [comment]
SPECFEM3D\_GLOBE has the [laplacian_smoothing_sem.F90](https://github.com/SPECFEM/specfem3d_globe/blob/devel/src/tomography/postprocess_sensitivity_kernels/laplacian_smoothing_sem.F90) implemented by Stephen Beller (@1sbeller) and
Congyue Cui  (@icui) . I am not sure if there was a plan to port it to Cartesian.
